libp2p/Host.cpp

```diff
@@ -355,20 +312,42 @@ void Host::runAcceptor()
 				}
 			}
 			
-			if (!success && m_socket->is_open())
+			if (!success && s->is_open())
```

Comment:
This code path could result from an exception being caught above. I'm not sure if it is still safe to access `s`.